docs/how-to/hip_runtime_api/memory_management/host_memory.rst

```diff
@@ -0,0 +1,205 @@
+* Pinned memory
+* Unified Memory
+
+These different types of memory should be used at different use cases. The
+pageable and pinned memory using explicit memory management, where the
+developers have direct control over memory operations, while at the unified
+memory case the developer gets a simplified memory model with less control over
+low level memory operations.
+
+The data transfer differences between the pageable or pinned memory usage
+represented in the next figure.
+
+.. figure:: ../../../data/how-to/hip_runtime_api/memory_management/pageable_pinned.svg
+
+The unified memory management and explicit memory management main difference
+highlighted at the following figure.
+
+.. figure:: ../../../data/how-to/hip_runtime_api/memory_management/unified_memory/um.svg
+
+The unified memory management described at the :doc:`/how-to/hip_runtime_api/memory_management/unified_memory`.
+
+Pageable memory
+================================================================================
+
+Pageable memory is exists on "pages" (blocks of memory), which can be
+migrated to other memory storage. For example, migrating memory between CPU
+sockets on a motherboard, or a system that runs out of space in RAM and starts
+dumping pages of RAM into the swap partition of your hard drive.
+
+Pageable memory is usually gotten when calling ``malloc`` or ``new`` in a C++
+application. The following example shows the pageable host memory usage in HIP.
+
+.. code-block:: cpp
+
+  #define ELEMENT_NUMBER 100
+
+  int *host_input, *host_output;
+  // Host allocation
+  host_input  = new int[ELEMENT_NUMBER];
+  host_output = new int[ELEMENT_NUMBER];
+
+  // Host data preparation
+  for (int i = 0; i < ELEMENT_NUMBER; i++) {
+    host_input[i] = i;
+  }
+  memset(host_output, 0, ELEMENT_NUMBER * sizeof(int));
+
+  int *device_input, *device_output;
+
+  // Device allocation
+  hipMalloc((int **)&device_input, ELEMENT_NUMBER * sizeof(int));
+  hipMalloc((int **)&device_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Device data preparation
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyHostToDevice);
+  hipMemset()
+
+  // Run the kernel
+  ...
+
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyDeviceToHost);
+
+  // Free host memory
+  delete[] host_input;
+  delete[] host_output;
+
+  // Free device memory
+  hipFree()
+  hipFree()
+
+.. note::
+
+  :cpp:func:`hipMalloc` and :cpp:func:`hipFree` are blocking calls, however, HIP
+  recently added non-blocking versions :cpp:func:`hipMallocAsync` and
+  :cpp:func:`hipFreeAsync` which take in a stream as an additional argument.
+
+Pinned memory
+================================================================================
+
+Pinned memory (or page-locked memory, or non-pageable memory) is host memory
+that is mapped into the address space of all GPUs, meaning that the pointer can
+be used on both host and device. Accessing host-resident pinned memory in device
```

Comment:
This sounds like it is confusing it with unified memory. Pinned memory is not necessarily accessable by the same pointer on the host and the device, and is also not necessarily mapped to the address space of the GPU. I thought main reason for using pinned memory was to improve performance of host-device transfers, as DMA is only possible from pinned memory, so usually the runtime has to transfer pageable memory to a pinned memory location and only then can copy it to the device.

Pinned memory mainly means, that it can not be swapped out, and actually physically resides in RAM, if my understanding is correct

Reply:
Discuss at separate issue.